keras_nlp/layers/transformer_decoder_test.py

```diff
@@ -266,56 +266,49 @@ def test_mask_propagation_without_cross_attention(self):
         outputs = decoder(decoder_sequence)
         self.assertAllEqual(outputs._keras_mask, mask)
 
-    def test_cached_decoding_is_correct(self):
+    @parameterized.named_parameters(
+        ("graph", False),
+        ("eager", True),
+    )
+    def test_cached_decoding_is_correct(self, eager):
         batch_size = 2
         seq_len = 5
         num_heads = 2
-        hidden_dim = 8
+        head_dim = 4
 
         layer = transformer_decoder.TransformerDecoder(
             intermediate_dim=4,
             num_heads=num_heads,
         )
-        x = tf.random.uniform(shape=[batch_size, seq_len, hidden_dim])
-        cache = tf.zeros(
-            [batch_size, 2, seq_len, num_heads, hidden_dim // num_heads]
-        )
-        # Build the intial cache.
-        initial_seq_len = 2
-        initial_inputs = x[:, :initial_seq_len, :]
+        x = tf.random.uniform(shape=[batch_size, seq_len, num_heads * head_dim])
+        cache = tf.zeros([batch_size, 2, seq_len, num_heads, head_dim])
         outputs = tf.zeros_like(x)
-        output, cache = layer(
```

Comment:
[optional] Although we have unified the first call with other calls, shall we keep the testing code? Like for the first call we slice until some position, like `x[:5]`. My intention is to mock the actual situation, where the first call has `seq_len > 1`, with following calls having `seq_len == 1`

Reply:
I think the logic of testing that you could pass an entire "chunk" to seed a cache is encapsulated well in the "no loop" call we compare with. I would focus on brevity while actually testing the dimension we want here.